pkg/pki/certmanagerpki/certmanager_user.go

```diff
@@ -206,8 +206,14 @@ func (c *certManager) getCA(user *v1alpha1.KafkaUser) (caName, caKind string) {
 		caKind = issuerRef.Kind
 	} else {
 		caKind = certv1.ClusterIssuerKind
-		caName = fmt.Sprintf(pkicommon.BrokerIssuerTemplate, c.cluster.Name)
+		caName = fmt.Sprintf(pkicommon.BrokerClusterIssuerTemplate,
+			c.cluster.Namespace, c.cluster.Name)
+	}
+	//Check if the new cluster issuer with namespaced name exists if not fall back to original one
+	var issuer *certv1.ClusterIssuer
+	err := c.client.Get(context.Background(), types.NamespacedName{Namespace: metav1.NamespaceAll, Name: caName}, issuer)
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
If the error is different from NotFound should that be returned upstream to trigger a reconcile?

Reply:
No, since this code only creates certificates resource for cert-manager. If the used issuer is not available yet Cert manager will retry. On the other hand the issuer creation error is handled by the Kafka Operator. If something happens we are retrying that.